models/base.rb

```diff
@@ -74,11 +74,27 @@ def inspect
       as_json
     end
 
+    def duplicate_safe(options = {})
+      object_hash = self.as_json
+
+      options.each{ |key, value|
+        next unless object_hash.key?(key) && self.class.default_attributes[key] != value
+
+        object_hash[key] = value
+      }
+      self.class.create(object_hash, options)
+    end
+
     def as_json(options = {})
```

Comment:
Noting so that we don't forget: Implementing `to_hash` might be even better than `as_json` so that we use the default `hash.as/to_json` functionality.

Reply:
This doesn't work as I expected. We decided to left it aside.